packages/test/test-version-utils/src/compatUtils.ts

```diff
@@ -207,6 +290,28 @@ export async function getCompatVersionedTestObjectProviderFromApis(
 		apis.dataRuntimeForLoading,
 	);
 
+	// We want to ensure that we are testing all latest rutime features, but only if both runtimes
+	// (one that creates containers and one that loads them) are supported them.
+	//
+	// Theoretically it should be fine to use config for apis.containerRuntimeForLoading?.version.
+	// If it's higher then apis.containerRuntime, then unknown to lower version of apis.containerRuntime
+	// would be ignored.
+	//
+	// But TestObjectProviderWithVersionedLoad.createLoader() implementation is dumb - it resets this.useCreateApi
+	// on first call and thus uses apis.containerRuntimeForLoading for any container created after.
+	// Many use non-first container instance to send ops, so that screws things up.
+	//
+	// As result, we absolutly need to use the min between two versions!
+	const versionForCreating = apis.containerRuntime?.version;
+	assert(versionForCreating !== undefined, "versionForCreating");
+	const versionForLoading = apis.containerRuntimeForLoading?.version;
+	assert(versionForLoading !== undefined, "versionForLoading");
+
+	const minVersion =
+		semver.compare(versionForCreating, versionForLoading) < 0
+			? versionForCreating
+			: versionForLoading;
```

Comment:
@andre4i - FYI: This change to use min version (instead of earlier using versionForLoading, as prior iteration did) removed all hits of 0x162 assert. Which suggests that very likely what we observed in production (when this assert was hit) was very similar kind of mismatch in expectations - a version of runtime used a feature (compression, op grouping - you name it) that the older version did not understand. But it did not fail, it simply ignored the op and moved forward. Which likely led to not creating data store, and thus further ops hitting this assert.

This underscores a need to build ASAP mechanism that allows such features to fail in predictable way in old runtimes. I.e. feature that detects "unknown file schema" and bails out long before it starts processing ops it does not understands.

If you can, can you please mention any ICM tickets or related stuff here please?

Reply:
> If you can, can you please mention any ICM tickets or related stuff here please?

I will send them to you in private